cpp/src/arrow/type_test.cc

```diff
@@ -364,152 +365,341 @@ TEST(TestField, TestMerge) {
   }
 }
 
-TEST(TestFieldPath, Basics) {
-  auto f0 = field("alpha", int32());
-  auto f1 = field("beta", int32());
-  auto f2 = field("alpha", int32());
-  auto f3 = field("beta", int32());
-  Schema s({f0, f1, f2, f3});
+struct FieldPathTestCase {
```

Comment:
Given that this is adding quite a lot of test code, and this file is already long, I think it may be time to move the `Field{Ref,Path}` tests to a separate test module.